dts/bindings/fpga/microchip,mpfs-fpga.yaml

```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2024 Microchip Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: Microchip Polarfire SOC FPGA driver
+
+compatible: "microchip,mpfs-fpga"
```

Comment:
I'm not all that keen on creating an "mpfs-fpga" compatible/binding as "mpfs-fpga" does not correspond to any actual hardware, but rather a single service provided by the system controller.

Ideally, this would look like what is currently used in Linux (and we are currently n the process of moving to for U-Boot) where the system controller is modelled as a mailbox. It looks something like:

```
	syscontroller: syscontroller {
		compatible = "microchip,mpfs-sys-controller";
		microchip,bitstream-flash = <&sys_ctrl_flash>;
		mboxes = <&mbox 0>;
	};

		mbox: mailbox@37020000 {
			compatible = "microchip,mpfs-mailbox";
			reg = <0x0 0x37020000 0x0 0x58>, <0x0 0x2000318C 0x0 0x40>,
			      <0x0 0x37020800 0x0 0x100>;
			interrupt-parent = <&plic>;
			interrupts = <96>;
			#mbox-cells = <1>;
			status = "disabled";
		};
```
Where the former is the mailbox client and the latter the mailbox provider.

Given only one service is being used in this PR, I suppose it would be viable to bind the driver for the fpga manager (funnily enough, we got declined from the fpga manager framework in Linux as there's no dynamic reconfiguration going on here) against the mailbox node for now. When later more services are implemented (eg hwrng), expand to have a proper provider/consumer driver setup? The only wrinkle is that the flash device used has to be got via the "microchip,bitstream-flash" property in the consumer node, not the provider node. Maybe Zephyr thinks differently, but node labels are not an ABI elsewhere, so the current code relying on the node label to get the flash device when a binding already exists with a dedicated property for this purpose does not seem desirable to me.